anime/components/homepage/HomePage.vue

```diff
@@ -1,407 +1,368 @@
-                    delete anime.aired_from;
-                    delete anime.aired_to;
-                    delete anime.summary;
-                    delete anime.anime_studio;
-                    delete anime.anime_genre;
-                    delete anime.number_rating;
-                    delete anime.anime_characters;
-                    delete anime.anime_awards;
-                    delete anime.season;
-                    delete anime.avg_rating;
-                    delete anime.id;
+			userStore.airingAnime = airingAnimeArr;
 
-                    return true;
-                });
+			airingAnime.value = userStore.airingAnime.slice(
+				userStore.startPageIndex,
+				userStore.endPageIndex
+			);
 
-                const airingAnime = [] as any;
+			loading.value = false;
+		})
+		.catch((err) => {
+			console.log(err);
+		});
+});
 
-                refineData.filter(function (anime: any) {
-                    if (anime.status == "Currently Airing") {
-                        airingAnime.push(anime);
-                    }
-                });
-
-                userStore.allAnime = refineData;
-                userStore.currentAnime = airingAnime;
-
-                pagePopularAnime.value = userStore.currentAnime.slice(
-                    userStore.startPageIndex,
-                    userStore.endPageIndex
-                );
-
-                loading.value = false;
-                console.log(userStore.currentAnime);
-            })
-            .catch((err) => {
-                console.log(err);
-            });
-    }
-    //}
-);
-
-function next() {
-    if (userStore.endPageIndex < userStore.currentAnime.length) {
-        userStore.startPageIndex += 11;
-        userStore.endPageIndex += 11;
-        userStore.pageNumber += 1;
-        pageExistLeft.value = true;
-        pagePopularAnime.value = userStore.currentAnime.slice(
-            userStore.startPageIndex,
-            userStore.endPageIndex
-        );
-    } else {
-        pageExistRight.value = false;
-    }
+function next(): void {
+	if (userStore.endPageIndex < userStore.airingAnime.length) {
+		userStore.startPageIndex += 11;
+		userStore.endPageIndex += 11;
+		userStore.pageNumber += 1;
+		pageExistLeft.value = true;
+		airingAnime.value = userStore.airingAnime.slice(
+			userStore.startPageIndex,
+			userStore.endPageIndex
+		);
+	} else {
+		pageExistRight.value = false;
+	}
 }
 
-function previous() {
-    if (userStore.pageNumber == 1) {
-        pageExistLeft.value = false;
-        pageExistRight.value = true;
-    } else {
-        userStore.startPageIndex -= 11;
-        userStore.endPageIndex -= 11;
-        userStore.pageNumber -= 1;
-        pageExistRight.value = true;
-        pagePopularAnime.value = userStore.currentAnime.slice(
-            userStore.startPageIndex,
-            userStore.endPageIndex
-        );
-    }
+function previous(): void {
+	if (userStore.pageNumber == 1) {
+		pageExistLeft.value = false;
```

Comment:
tbh do you really need pageExistLeft if the page val is trying to go less then 0 dont allow it and same concept applys for the right(you can calc by dividing the num per page by total anime.)